src/vcbc/mod.rs

```diff
@@ -0,0 +1,58 @@
+use crate::{crypto::public::PubKey, ProposalService};
+
+use self::payload::Message;
+
+mod deliver;
+mod echo;
+mod error;
+pub  mod log;
+mod payload;
+mod propose;
+
+pub trait State {
+    fn process_message(self: Box<Self>, log: &log::Log) -> Box<dyn State>;
+}
```

Comment:
Interesting idea to use a state trait :+1:, I would usually model this with a state enum, but curious to see how this approach looks later on.